app/src/main/java/org/kiwix/kiwixmobile/main/KiwixWebViewClient.java

```diff
@@ -91,6 +92,10 @@ public void onReceivedError(WebView view, int errorCode, String description, Str
 
   @Override
   public void onPageFinished(WebView view, String url) {
+    if (BuildConfig.IS_CUSTOM_APP && url.equals("content://" + BuildConfig.APPLICATION_ID + ".zim.base/null")) {
+      Log.e(TAG_KIWIX, "Abandoning WebView as there's a problem getting the content for this custom app.");
+      return;
+    }
     if ((url.equals("content://" + BuildConfig.APPLICATION_ID + ".zim.base/null"))
         && !BuildConfig.IS_CUSTOM_APP) {
```

Comment:
we could change this if statement and inside it have different branches based on `IS_CUSTOM_APP`, DRYer

Reply:
And I would rather throw an exception and just crash the app than return and continue execution flow.
That is if my assumption is correct that the zim should always be there for custom?

Reply:
@macgills From what I understand we (the app) shouldn't assume the content will always be on the device _even if the app was installed by the user from Google Play Store_. And for Kiwix apps many are installed using other approaches e.g. users sharing files (about 1/3 of the crash data in Android Vitals for Kiwix indicates the users installed the app from other sources, and that's only a subset who have their devices automatically sending usage and diagnostics to Google). Google provides example code for app developers to add to their app to download the file from Google Play, and we could simply implement this, however it then makes the app and therefore the user effectively dependent on having a working internet connection, a Google account, etc. From discussions with @mhutti1 yesterday we agree we'd like to design and implement a practical approach for the app to obtain the files (possibly where users copy the expansion file (the ZIM file) in the same way they obtained the app). In summary, the app should cope elegantly with the lack of ZIM file and inform the user without crashing. Meanwhile I'll test the app's behaviour (particularly whether it displays the message correctly and doesn't crash) if I modify this code to throw an exception.

Reply:
Re https://github.com/kiwix/kiwix-android/pull/1365#discussion_r313486523 I'm happy to improve the DRY ness of the code and will have a go at doing so. Thanks for the suggestion.

Reply:
Thanks for the info, my assumption was incorrect.

Exception unnecessary.

```
if (badUrl){
  if(isCustomApp){}
  else{}
}
```
will resolve this conversation